internal/controller/isbservicerollout_controller.go

```diff
@@ -615,9 +615,9 @@ func (r *ISBServiceRolloutReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Group:   common.NumaflowAPIGroup,
 		Version: common.NumaflowAPIVersion,
 	})
-	if err := controller.Watch(source.Kind(mgr.GetCache(), isbServiceUns),
-		handler.EnqueueRequestForOwner(mgr.GetScheme(), mgr.GetRESTMapper(), &apiv1.ISBServiceRollout{}, handler.OnlyControllerOwner()),
-		predicate.ResourceVersionChangedPredicate{}); err != nil {
+	if err := controller.Watch(source.Kind(mgr.GetCache(), isbServiceUns,
+		handler.TypedEnqueueRequestForOwner[*unstructured.Unstructured](mgr.GetScheme(), mgr.GetRESTMapper(),
+			&apiv1.ISBServiceRollout{}, handler.OnlyControllerOwner()))); err != nil {
```

Comment:
I'm trying to add `predicate.ResourceVersionChangedPredicate{}` according to new definition of watch